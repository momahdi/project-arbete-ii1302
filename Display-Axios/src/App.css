.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
.screen{
 background-color: #d3d3d3;

  width: 70%;
  height: 70vh;
  border-radius:  30px;
  display: flex;
 
  color: black;
  -webkit-box-shadow: 0px 0px 14px 3px rgba(0,0,0,0.75);
-moz-box-shadow: 0px 0px 14px 3px rgba(0,0,0,0.75);
box-shadow: 0px 0px 14px 3px rgba(0,0,0,0.75);
  
}
.screen {
  display: grid;
  grid-template-columns: repeat(2, 1fr);

  grid-auto-rows: minmax(100px, auto);
}

.currenttime{
  grid-column: 1 ;
  grid-row: 1;
width: 100%;
height: 100%;
border-right:solid;
border-color: #282c34;
color:#282c34;
font-weight: bold;

}
.senttime{

  grid-column: 2 ;
  grid-row: 1 ;
  width: 100%;
  height: 100%;
  border-left:solid;
  color:#282c34;
  border-color: #282c34;
  font-weight: bold;
}
.text{
  grid-column: 1 / 3 ;
  grid-row: 2 / 4 ;
  width: 100%;
  height: 100%;
  border-top:solid;
  color:#282c34;
  border-color: #282c34;
  font-weight: bold;
}

#input{
  width: 50%;
  margin-bottom: 30px;
  border-radius:  30px;
  padding: 10px 40px;
  border: none;

}
.send{
  position: absolute;
  background-color: transparent;
  border: none;
}


.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
