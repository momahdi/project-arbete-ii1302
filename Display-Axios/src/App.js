
import React, { Component } from 'react';
import './App.css';
const axios = require('axios');


class App extends Component{
  constructor(props){
    super(props);
    this.state={
      text: "",
      currenttime:new Date().toLocaleString().slice(12.0),
      senttime:"",
      displaylink:"https://iot-display.herokuapp.com/display/get/",
      displayid: ""
      
    }
  }


  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    },this.checkFields);
  };


componentWillMount(){
setInterval(() => {
  getu(this.state.displaylink + this.state.displayid)
  var test = stateModule.getState();
 
  if(test!==this.state.text&& test!== undefined){
    var event = {
      target: {
        name: "senttime",
        value: new Date().toLocaleString().slice(12.0)
      }
    };
    this.handleChange(event)

  }

  var event = {
    target: {
      name: "text",
      value: test
    }
  };
  this.handleChange(event)
}, 10000);}

  
componentDidMount(){
  this.intervalID = setInterval(
    () => this.tick(),
    1000
  );
}
  tick() {
    var event = {
      target: {
        name: "currenttime",
        value: new Date().toLocaleString().slice(12.0)
      }
    };
    this.handleChange(event)

  }

  handlechange=(e)=>{
    this.setState({displayid:e.target.value})

}
handleSubmitbtn=(e)=>{
  this.setState({text:"Connecting to Display"})
  console.log("hej")
}
onKeyPress(event) {
  if (event.which === 13 /* Enter */) {
    event.preventDefault();
  }
}
    
render(){
  
  return (
    <div className="App">
      <header className="App-header">

                <h5>Enter Diplay ID</h5>
               < form onKeyPress={this.onKeyPress}>
                    <input id="input" onChange ={ this.handlechange} 
                        value={this.state.displayid}
                        placeholder="Enter ID"
                        type="text"/>
                        <button type ="button" onClick={e=>this.handleSubmitbtn()} className="send"> 
                        <img  width="35"
                            src ={ require('./pictures/search.png')}
                            alt ={"could not load"}/>
                        </button>
                        </form>
                        
            
      




        <div className="screen">
        <div className = "text ">
          <p></p> 
        <div id="text">{this.state.text}</div>
        </div>
        <div className = "currenttime ">
        <p>Current Time</p> 
          <div id="">{this.state.currenttime}
          </div>
          </div>
        <div className = "senttime ">
        <p>Recieved Time</p> 
          <div id="">{this.state.senttime}
          </div>
          </div>
        </div>
      </header>
    </div>
  );
}
}
export default App;


const getu = async url => {
  try {
    const response = await axios.get(url);
      var  myData = myData = response.data.display.message.text;
      stateModule.changeState(myData)
  } catch (error) {
    console.error(error);
  }
};

var stateModule = (function () {
  var state; // Private Variable

  var pub = {};// public object - returned at end of module

  pub.changeState = function (newstate) {
      state = newstate;
  };

  pub.getState = function() {
      return state;
  }

  return pub; // expose externally
}());




