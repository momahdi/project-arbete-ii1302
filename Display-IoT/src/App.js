
import React, { Component } from 'react';
import './App.css';

var IoTDevice = require ('./device/device.js');

/* Edit these lines to reflect your IoT platform config. */
const ORG_ID = "cl3hm5";
const ACCESS_TOKEN = "secure_token_ibm";
var device = new IoTDevice(ORG_ID, ACCESS_TOKEN);


class App extends Component{
  constructor(props){
    super(props);
    this.state={
      text: "",
      currenttime:new Date().toLocaleString().slice(12.0),
      senttime:""

    }
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    },this.checkFields);
  };


componentWillMount(){
setInterval(() => {
  /* This pushes / publishes data to Watson IoT*/
  device.Push('getCurrentMessage');
  var test = device.getMessage()
  if(test!==this.state.text&& test !== undefined){
    var event = {
      target: {
        name: "senttime",
        value: new Date().toLocaleString().slice(12.0)
      }
    };
    this.handleChange(event)
  }
  console.log(test)
  console.log(device.IsConnected())
  var event = {
    target: {
      name: "text",
      value: test
    }
  };
  this.handleChange(event)
}, 10000);}

  
componentDidMount(){
  this.intervalID = setInterval(
    () => this.tick(),
    1000
  );
}

  tick() {
    var event = {
      target: {
        name: "currenttime",
        value: new Date().toLocaleString().slice(12.0)
      }
    };
    this.handleChange(event)

  }
    
render(){
  
  return (
    <div className="App">
      <header className="App-header">
     
        <div className="screen">
        <div className = "text ">
          <p>Message:</p> 
        <div id="text">{this.state.text}</div>
        </div>
        <div className = "currenttime ">
        <p>Current Time</p> 
          <div id="">{this.state.currenttime}
          </div>
          </div>
        <div className = "senttime ">
        <p>Sent Time</p> 
          <div id="">{this.state.senttime}
          </div>
          </div>
        </div>
      </header>
    </div>
  );
}
}

export default App;
