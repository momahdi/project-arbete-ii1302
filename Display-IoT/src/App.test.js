import { create } from "react-test-renderer";
import React from "react";
import App from "./App.js"
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
describe('device module',()=>{
  let wrapper;
  let mockSubmit;
  beforeEach(() => {
    mockSubmit = jest.fn();
    wrapper = shallow(<App submit={mockSubmit} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleChange", () => {
    it("should call setState on title", () => {
      const mockEvent = {
        target: {
          name: "text",
          value: "test"
        }
      };

      const expected = {
        text: "test",
        currenttime:new Date().toLocaleString().slice(12.0),
        senttime:""
       
      };
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state()).toEqual(expected);
    });
  });

})