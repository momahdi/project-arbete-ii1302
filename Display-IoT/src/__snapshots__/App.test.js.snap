// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`device module should match the snapshot 1`] = `ShallowWrapper {}`;
